java/src/main/java/io/cucumber/java/GlueAdaptor.java

```diff
@@ -54,7 +54,12 @@ void addDefinition(Method method, Annotation annotation) {
         } else if (annotation.annotationType().equals(DefaultParameterTransformer.class)) {
             glue.addDefaultParameterTransformer(new JavaDefaultParameterTransformerDefinition(method, lookup));
         } else if (annotation.annotationType().equals(DefaultDataTableEntryTransformer.class)) {
-            glue.addDefaultDataTableEntryTransformer(new JavaDefaultDataTableEntryTransformerDefinition(method, lookup));
+            if (method.getAnnotation(DefaultDataTableEntryTransformer.class).titleCaseHeaders()) {
```

Comment:
Rather then introducing a new class it might be better to add a constructor parameter to the existing `JavaDefaultDataTableEntryTransformerDefinition`. It avoids the need to create a subclass and duplicate all the tests for the original class and the sub class.